core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -3148,9 +3148,7 @@ private void checkPushJoinThroughUnionOnRightDoesNotMatchSemiOrAntiJoin(JoinRelT
         + "from t1\n"
         + "where (t1.a, t1.y) in ((1, 2), (3, null), (7369, null), (7499, 30), (null, 20), (null, 5))";
     sql(sql)
-        .withRelBuilderConfig(b -> b.withSimplifyValues(false))
-        .withRule(CoreRules.PROJECT_VALUES_MERGE,
-            CoreRules.UNION_TO_VALUES)
+        .withRule(CoreRules.FILTER_SUB_QUERY_TO_CORRELATE)
```

Comment:
are these tests testing the same thing as the original tests?
if not, perhaps you should add new tests rather than modifying the original ones